static/app/components/forms/form.tsx

```diff
@@ -80,6 +80,34 @@ export interface FormProps
   submitPriority?: ButtonProps['priority'];
 }
 
+function getSubmitButtonTitle(form: FormModel) {
+  if (form.isFormIncomplete) {
+    return t('Required fields must be filled out');
+  }
+
+  if (form.isError) {
+    const errors = form.getErrors();
+    const hasRequiredFieldError = [...errors].some(
+      ([_field, message]) => message === fieldIsRequiredMessage
+    );
+
+    if (hasRequiredFieldError) {
+      const allRequiredFieldErrors = [...errors].every(
+        ([_field, message]) => message === fieldIsRequiredMessage
+      );
+
+      if (allRequiredFieldErrors) {
+        return t('Required fields must be filled out');
+      }
+
+      return t('Required fields must be filled out and inputs must be valid');
+    }
+
+    return t('Fields must contain valid inputs');
+  }
+  return undefined;
```

Comment:
nit (sorry should have commented earlier): you can use guard statements to reduce the nesting here

```suggestion
  if (!form.isError) {
    return undefined;
  }
  
  const errors = form.getErrors();
  const hasRequiredFieldError = [...errors].some(
    ([_field, message]) => message === fieldIsRequiredMessage
  );

  if (hasRequiredFieldError) {
    const allRequiredFieldErrors = [...errors].every(
      ([_field, message]) => message === fieldIsRequiredMessage
    );

    if (allRequiredFieldErrors) {
      return t('Required fields must be filled out');
    }

    return t('Required fields must be filled out and inputs must be valid');
  }

  return t('Fields must contain valid inputs');
```